src/addrman.cpp

```diff
@@ -567,6 +569,13 @@ void CAddrMan::ResolveCollisions_()
                         Good_(info_new, false, GetAdjustedTime());
                         erase_collision = true;
                     }
+                } else if (GetAdjustedTime() - info_new.nLastSuccess > ADDRMAN_TEST_WINDOW) {
+                    // If the collision hasn't resolved in some reasonable amount of time,
+                    // just evict the old entry -- we must not be able to
+                    // connect to it for some reason.
+                    LogPrint(BCLog::ADDRMAN, "Unable to test; swapping %s for %s in tried table anyway\n", info_new.ToString(), info_old.ToString());
```

Comment:
I'm not sure, but shouldn't the variables be swapped?

Reply:
The word "swap" confused me, also in the log message a few lines above (`if (GetAdjustedTime() - info_old.nLastTry > 60)`). The phrase "replace [old] with [new]" is more clear.

Also, this new behaviour contradicts the "when in doubt, don't evict" rule I mentioned above. Though maybe I'm wrong there?  It seems more consistent to just drop the entry.

Reply:
The idea is that if we've been trying to test the old entry and can't for some reason, then something is probably wrong with that old entry, so we do want to do the replacement.

Reply:
I can edit the language of the log message if this is confusing.